app/models/user.rb

```diff
@@ -46,6 +47,12 @@ def account_id
     @account_id ||= account&.id
   end
 
+  def credential_lock
+    return @credential_lock unless @credential_lock.nil?
+
+    @credential_lock ||= user_verification&.locked == true
```

Comment:
just do `@credential_lock = user_verification&.locked`, the memoization syntax sugar isn't necessary since you're doing the `return` above, and no need to compare to `true`